### PR TITLE
Created Lesson service interface + implementation

### PR DESCRIPTION
**Pull request by Fuyunori**
Currently still a work-in-progress, checklist of things to-do:
- [x] interface
- [x] implementation

### CODE REVIEW

**Review by paveltobias — APPROVED** [review]

File: service/src/main/java/tennisclub/service/LessonService.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+public interface LessonService {
+    void create(Lesson lesson);
+    Lesson update(Lesson lesson);
+    void remove(Lesson lesson);
+    List<Lesson> listByStartTime(LocalDateTime startTime);
+    List<Lesson> listByEndTime(LocalDateTime endTime);
+    List<Lesson> listByTimeInterval(LocalDateTime from, LocalDateTime to);
+    List<Lesson> listByCapacity(Integer capacity);
+    List<Lesson> listByLevel(Level level);
+}
```
Comment (mazanec202) [inline service/src/main/java/tennisclub/service/LessonService.java:19]: I have a feeling there should be also findById(), but I guess you decided to add it later as it is not part of the "mechanical" services we defined, right?:)

**Review by mazanec202 — APPROVED** [review]

File: service/src/main/java/tennisclub/service/LessonService.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package tennisclub.service;
+
+import tennisclub.entity.Lesson;
+import tennisclub.entity.User;
+import tennisclub.entity.enums.Level;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
```
Comment (mazanec202) [inline service/src/main/java/tennisclub/service/LessonService.java:9]: In the reference eshop project they used the @service annotation also for the interface, I guess we should add it here as well.

**Review by mazanec202 — APPROVED** [review]

File: service/src/main/java/tennisclub/service/LessonService.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+public interface LessonService {
```
Comment (mazanec202) [inline service/src/main/java/tennisclub/service/LessonService.java:10]: In the reference eshop project they used the @service annotation also for the interface, I guess we should add it here as well.

**Review by mazanec202 — APPROVED** [review]

File: service/src/main/java/tennisclub/service/LessonService.java (hunk 1 above)
Comment (Fuyunori) [reply service/src/main/java/tennisclub/service/LessonService.java:19]: Thank you for pointing this out 😄

File: service/src/main/java/tennisclub/service/LessonService.java (hunk 3 above)
Comment (Fuyunori) [reply service/src/main/java/tennisclub/service/LessonService.java:10]: In the reference project it's confusing 🤔 sometimes it's there, sometimes not 😱

**Pull request merged**